scene/2d/tile_map_layer.cpp

```diff
@@ -1670,10 +1670,15 @@ void TileMapLayer::_internal_update(bool p_force_cleanup) {
 		dirty.flags[i] = false;
 	}
 
-	// List the cells to delete definitely.
+	// List the cells to delete definitely, and all dirty cell's positions to notify script of cell updates.
 	Vector<Vector2i> to_delete;
+	TypedArray<Vector2i> dirty_cell_positions;
+	bool script_update_callback = cells_initialized && !p_force_cleanup && dirty.cell_list.first() && GDVIRTUAL_IS_OVERRIDDEN(_update_cells);
```

Comment:
I fell like this function would mostly be used to react to when the user changes the tile map (in editor or through `set_cell`), when it would run some custom checks to ensure that the tiles are placed correctly (this is the use case presented in godotengine/godot-proposals#668), or to have some custom auto-tile algorithm (this is the way I intend to use it). Those checks and algorithms may be expensive, therefore it may be desirable for these algorithms not to be run on intialization and during cleanup updates (those updates are currently applied to every single cell in the tilemap, meaning that such algorithms could take minutes to run depending on the tile map's size).

I do see the value with having this function exactly follow the internals though. However, I think that if it does follow the internal updates, there should be a way to know if the TileMapLayer has been initialized or is being cleaned up (to be able to ignore those updates). As far as i'm aware, the only way to ignore them would currently be to do:
```gdscript
var initialized: bool = false
var pre_delete: bool = false

func _notification(what: int) -> void:
    if what == NOTIFICATION_PREDELETE:
        pre_delete = true

func _update_cells(coords: Array[Vector2i]) -> void:
    if not initialized or pre_delete:
        initialized = true
        return
    # ...
```

Reply:
Yeah well, I believe in that case it's better to let the user decide whether it is worth running an update or not, so I think we should follow the internals here. I understand your use case is to react to cell changes, but other might just want to use that to implement their own runtime operations.

We could maybe provide more arguments to `_update_cells()`. What I suggest is that you do is imitate what is done in other update functions. Something like this:
```
void TileMapLayer::_update_cells_callback(bool p_force_cleanup) {
	// Check if we should cleanup everything.
	bool forced_cleanup = p_force_cleanup || !enabled || !is_inside_tree() || tile_set.is_null();
	
	// Call the callback with `dirty_cell_positions` and `forced_cleanup`
}
```
Then for initialization, it will bu up to the user to detect the first call to `_update_cells`

Reply:
Ok, makes sense ! I'll do that !